### PR TITLE
nitrokey-app: init at 0.4.0

### PR DESCRIPTION
**Pull request by KaiHa**
###### Motivation for this change

The nitrokey-app is useful tool for everyone who uses a Nitrokey Pro or a Nitrokey Storage.
###### Things done
- [X] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

See https://github.com/Nitrokey/nitrokey-app and
https://www.nitrokey.com/download for information about the
nitrokey-app.

### CODE REVIEW

**Comment by KaiHa** [comment]
I have seen that the Travis build has failed with a checksum mismatch, but I do not understand why because on my machine `nix-prefetch-git` reports the checksum I have put in the `default.nix`:
```
 $ nix-prefetch-git --out ./foo https://github.com/Nitrokey/nitrokey-app.git 47fe44bc0a9ef42cb939443dbba45bf57837caae                  
Initialized empty Git repository in /tmp/git-checkout-tmp-NR2mO9V7/nitrokey-app-47fe44b/.git/
fatal: Invalid refspec '+refs/tags/v0.4^{}'
remote: Counting objects: 3536, done.
remote: Total 3536 (delta 0), reused 0 (delta 0), pack-reused 3536
Receiving objects: 100% (3536/3536), 19.74 MiB | 854.00 KiB/s, done.
Resolving deltas: 100% (2447/2447), done.
From https://github.com/Nitrokey/nitrokey-app
 * [new branch]      coverage   -> origin/coverage
 * [new branch]      coverity-scan -> origin/coverity-scan
 * [new branch]      debian     -> origin/debian
 * [new branch]      get_app_version_from_repository -> origin/get_app_version_from_repository
 * [new branch]      issue_33_sync_for_not_encrypted -> origin/issue_33_sync_for_not_encrypted
 * [new branch]      issue_prevent_make_install_dirty_the_root_filesystem -> origin/issue_prevent_make_install_dirty_the_root_filesystem
 * [new branch]      master     -> origin/master
 * [new branch]      progress_bar_dialog_remaining_time -> origin/progress_bar_dialog_remaining_time
 * [new branch]      release    -> origin/release
 * [new branch]      travis_update -> origin/travis_update
 * [new branch]      tray_test  -> origin/tray_test
 * [new tag]         v0.1       -> v0.1
 * [new tag]         v0.2       -> v0.2
 * [new tag]         v0.3.2     -> v0.3.2
 * [new tag]         v0.4       -> v0.4
Switched to a new branch 'fetchgit'
removing `.git'...

git revision is 47fe44bc0a9ef42cb939443dbba45bf57837caae
path is /nix/store/693nmz96vssgxsy18sadmp67pyfyhz16-nitrokey-app-47fe44b
git human-readable version is v0.4
Commit date is 2016-07-02 23:30:58 +0200
hash is 1r3xpwnk959iz1fhs3517v28sm27rpczdlc5k4d72ncch0lpfqzh
{
  url = "https://github.com/Nitrokey/nitrokey-app.git";
  rev = "47fe44bc0a9ef42cb939443dbba45bf57837caae";
  sha256 = "1r3xpwnk959iz1fhs3517v28sm27rpczdlc5k4d72ncch0lpfqzh";
}
```

Can someone shed some light on this?

**Comment by jokogr** [comment]
You could try with `fetchFromGitHub` instead of `fetchgit`, I guess. Here's [an example](https://github.com/NixOS/nixpkgs/blob/5915716f134ecc3316983c59d48aea92d92f5c07/pkgs/applications/networking/syncthing/default.nix).

**Comment by fpletz** [comment]
@KaiHa Are you on 16.03? We changed the way hashes are generated after 16.03, so all fetchgit hashes have changed. If the package doesn't need the git repo, `fetchFromGitHub` would be preferred as fetching the source is faster and the source takes less storage.

**Comment by KaiHa** [comment]
Yes, you are right I am on 16.03. I will update the hashes and will also give `fetchFromGitHub` a try. Thanks for the clarification.

**Comment by Mic92** [comment]
any updates?

**Comment by KaiHa** [comment]
@Mic92 thanks for reminding me. I have updated the hash and use now `fetchFromGitHub`. Let's see what Travis thinks about it.

File: pkgs/tools/security/nitrokey-app/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  meta = {
+    description      = "Provides extra functionality for the Nitrokey Pro and Storage";
+    longDescription  = ''
+       The nitrokey-app provides a QT system tray widget with wich you can
+       access the extra functionality of a Nitrokey Storage or Nitrokey Pro.
+       See https://www.nitrokey.com/ for more information.
+    '';
+    homepage         = https://github.com/Nitrokey/nitrokey-app;
+    repositories.git = https://github.com/Nitrokey/nitrokey-app.git;
+    license          = stdenv.lib.licenses.gpl3;
```
Comment (Mic92) [inline pkgs/tools/security/nitrokey-app/default.nix]: do you want to maintain it?

**Comment by Mic92** [comment]
looks good otherwise.

File: pkgs/tools/security/nitrokey-app/default.nix (hunk 1 above)
Comment (KaiHa) [inline pkgs/tools/security/nitrokey-app/default.nix]: I think the effort will be manageable. So yes, I am willing to maintain this package.

Comment (Mic92) [inline pkgs/tools/security/nitrokey-app/default.nix]: ok, then add:

```
maintainer = stdenv.lib.maintainers.kaiha;
```

and 

you self to `lib/maintainers.nix`

Comment (KaiHa) [inline pkgs/tools/security/nitrokey-app/default.nix]: @Mic92  Sorry for the long delay, now I have added myself as maintainer of this package.

**Pull request merged**